nimble/host/services/gatt/src/ble_svc_gatt.c

```diff
@@ -122,6 +132,40 @@ ble_svc_gatt_cl_sup_feat_access(uint16_t conn_handle, uint16_t attr_handle,
     return 0;
 }
 
+static int
+ble_svc_gatt_db_hash_access(uint16_t conn_handle, uint16_t attr_handle,
+                            struct ble_gatt_access_ctxt *ctxt, void *arg)
+{
+    int rc;
+    struct ble_gap_sec_state sec_state;
+    const uint8_t *hash;
+    uint8_t local_hash[BLE_GATT_DB_HASH_SZ];
+
+    if (ctxt->op != BLE_GATT_ACCESS_OP_READ_CHR) {
+        return BLE_ATT_ERR_WRITE_NOT_PERMITTED;
+    }
+
+    hash = ble_gatts_get_db_hash();
+
+    for (int i = 0; i < BLE_GATT_DB_HASH_SZ; i++) {
```

Comment:
Declare the loop control variable at the beginning of the function with the rest of the variables.
```suggestion
    for (i = 0; i < BLE_GATT_DB_HASH_SZ; i++) {
```